src/main/java/javax/measure/unit/ProductUnit.java

```diff
@@ -69,17 +84,50 @@ private ProductUnit(Element[] elements) {
         _elements = elements;
     }
 
+    private static final LoadingCache<ProductUnitKey, Unit<? extends Quantity>> INSTANCE_CACHE =
+            Caffeine.newBuilder()
+                    .maximumSize(4096)
+                    .build(key -> doGetInstance(key.leftElems, key.rightElems));
+
+    private static final class ProductUnitKey {
+        private final Element[] leftElems;
+        private final Element[] rightElems;
+        private final int hashCode;
```

Comment:
minor: all the other variables that are related to optimizations (mostly storing computed values) start with underscore (debatable choice, but it precedes this PR). Consider renaming this to `_hashCode` for the sake of consistency.

Reply:
It's not just variables related to optimization, it's all fields. Like in the 90s :) I'll align my style.